File: objective-c/cpp_to_objc/SportsManagement/SportsManagement/SportsClient.cpp
//
//  SportsClient.cpp
//

#include <iostream>
#include "SportsClient.h"

using namespace std;

SportsClient::SportsClient()
{
    cout << "Starting record for client, 1..." << endl;
    this->firstName = "Anonymous";
    this->lastName = "Coward";
    this->email = "";
    this->age = 0;
}

SportsClient::SportsClient(string firstName, string lastName)
{
    cout << "Starting record for client, 2..." << endl;
    this->firstName = firstName;
    this->lastName = lastName;
    this->email = "";
    this->age = 0;    
}

SportsClient::SportsClient(string firstName, string lastName, string email, unsigned int age)
{
    cout << "Starting record for client, 3..." << endl;
    this->firstName = firstName;
    this->lastName = lastName;
    this->email = email;
    this->age = age;
}

string SportsClient::getFirstName()
{
    return firstName;
}

string SportsClient::getLastName()
{
    return lastName;
}

void SportsClient::setEmail (string email)
{
    this->email = email;
}

string SportsClient::getEmail()
{
    return email;
}

void SportsClient::setAge (unsigned int age)
{
    this->age = age;
}

unsigned int SportsClient::getAge()
{
    return age;
}

void SportsClient::setComments (string comments)
{
    this->comments = comments;
}

string SportsClient::getComments()
{
    return comments;
}

void SportsClient::sendEmail (string subject, string message)
{
    cout << "Sending email..." << endl;
}

void SportsClient::print()
{
    cout << firstName << " " << lastName << endl;
}

SportsClient::~SportsClient()
{
    cout << "Bye!" << endl;
}

File: examples/objective-c/cpp_to_objc/SportsManagementObjC/SportsManagementObjC/SportsClient.m
//
//  SportsClient.m
//  SportsMgmtObjC
//
//  Created by Ming Chow on 1/24/12.
//  Copyright (c) 2012 Tufts University. All rights reserved.
//

#import "SportsClient.h"

@implementation SportsClient

@synthesize firstName, lastName, email, age, comments;

-(id) init
{
    NSLog(@"Starting record for client, 1...");
    self = [super init];
    if (self) {
        firstName = @"";
        lastName = @"";
        email = @"";
        age = 0;
        comments = @"";
    }
    return self;
}

-(id) initWithFirstName: (NSString *) fname withLastName: (NSString *) lname
{
    NSLog(@"Starting record for client, 2...");
    self = [super init];
    if (self) {
        firstName = fname;
        lastName = lname;
        email = @"";
        age = 0;
        comments = @"";
    }
    return self;    
}

-(id) initWithFirstName: (NSString *) fname withLastName: (NSString *) lname withAge: (unsigned int) a withEmail: (NSString *) e
{
    NSLog(@"Starting record for client, 3...");
    self = [super init];
    if (self) {
        firstName = fname;
        lastName = lname;
        email = e;
        age = a;
        comments = @"";
    }
    return self;
}

-(void) sendEmailWithSubject: (NSString *) subject withMessage: (NSString *) message
{
    NSLog(@"Sending email...");
}

-(void) print
{
    NSLog(@"%@ %@", firstName, lastName);
}

-(void) dealloc
{
    NSLog(@"Bye!");
}

@end
